clippy_lints/src/empty_with_brackets.rs

```diff
@@ -70,10 +75,21 @@ declare_clippy_lint! {
     "finds enum variants with empty brackets"
 }
 
-declare_lint_pass!(EmptyWithBrackets => [EMPTY_STRUCTS_WITH_BRACKETS, EMPTY_ENUM_VARIANTS_WITH_BRACKETS]);
+enum Usage {
+    Unused(Span),
```

Comment:
Is it possible to avoid storing the `Span`s here and instead construct them later in `check_crate_post`? We should be able to get them back given just the constructor's `LocalDefId` in the map. https://github.com/rust-lang/rust-clippy/pull/12971#discussion_r1648531815